File: WebAPI/Models/AdvertModelForRequest.cs
using System;
using System.Collections.Generic;

namespace WebAPI.Models
{
    public class AdvertModelForRequest
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public DateTime EditDate { get; set; }
        public List<ImageModel> Images { get; set; }
    }
}

File: WebAPI/ClientApp/src/app/models/AdvertModelForRequest.ts
import { ImageModel } from "./ImageModel";

export class AdvertModelForRequest {
    public id: number;
    public name: string;
    public editDate: Date;
    public images: ImageModel[];

    public constructor(_id: number, _name: string, _editDate: Date, _images: ImageModel[]) {
        this.id = _id;
        this.name = _name;
        this.editDate = _editDate;
        this.images = _images;
    }
}